docs/proposals/default-service-plans.md

```diff
@@ -0,0 +1,544 @@
+# Default Service Plans in Service Catalog
+
+* [Abstract](#abstract)
+* [Summary](#summary)
+* [Motivation](#motivation)
+* [Goals and Non-Goals](#goals-and-non-goals)
+* [Storage Classes vs. Service Classes](#storage-classes-vs-service-classes)
+* [User Experience](#user-experience)
+  * [Cluster Operator](#cluster-operator)
+     * [Broker Installation](#broker-installation)
+     * [Cluster Customization](#cluster-customization)
+  * [Developer](#developer)
+     * [Service Class Discovery](#service-class-discovery)
+     * [Legacy Provisioning](#legacy-provisioning)
+     * [Provisioning with Defaults](#provisioning-with-defaults)
+     * [Provisioning by Service Type](#provisioning-by-service-type)
+     * [Service Instance Management](#service-instance-management)
+  * [Helm Charts](#helm-charts)
+     * [Broker](#broker)
+     * [Author](#author)
+     * [Consumer](#consumer)
+* [Implementation](#implementation)
+  * [Service Class](#service-class)
+  * [Service Plan](#service-plan)
+  * [Service Instance](#service-instance)
+  * [Service Binding](#service-binding)
+
+## Abstract
+
+Let's wholesale rip off how default storage classes for volumes work and apply
+it to Service Catalog! 😎 This allows us to specify a dependency on a type of
+service without requiring upfront knowledge of the underlying broker.
+
+> "My application requires a mysql database."
+
+> "I just want to try out this application using the recommended defaults."
+
+> "I need to encourage everyone to use the cheapest plans possible in our
+> integration environment by making a small database the default."
+
+## Summary
+
+1.  Add default provisioning and binding parameters to Service Classes.
```

Comment:
binding parameters could be unique per plan, makes sense to tie them to plans; same comment about provisioning parameters, perhaps the class just has a default plan pointer (or use the existing feature for default plan) and then allow a plan to say how it wants to be defaulted

Reply:
Originally I had the defaulting parameters defined on both the class and the plan, so that if all the plans for a class use the same defaults (which is the case for azure) the defaults didn't have to be redefined and managed in multiple places.

Perhaps allow for it on either, and if a plan doesn't have defaults set, it uses the defaults on the class?